models/openacademy_session.py

```diff
@@ -1,12 +1,48 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 class Session(models.Model):
     _name = "openacademy.session"
     _description = "This is the description of Session model."
 
     name = fields.Char()
-    start_date = fields.Date()
+    start_date = fields.Date(default=fields.Date.today)
     duration = fields.Integer()
     number_seats = fields.Integer("Number of seats")
-    instructor_id = fields.Many2one("res.partner", ondelete="restrict")
+    instructor_id = fields.Many2one("res.partner", domain="['|', ('instructor', '=', 'True'), ('category_id.name', '=', 'Teacher / Level 1'), ('category_id.name', '=', 'Teacher / Level 2')]", ondelete="restrict")
```

Comment:
- Split line, please
- `True` should be unquoted, otherwise it would be interpreted as string instead of boolean

```suggestion
    instructor_id = fields.Many2one(
        "res.partner",
        domain=[
            '|',
            ('instructor', '=', True),
            ('category_id.name', '=', 'Teacher / Level 1'),
            ('category_id.name', '=', 'Teacher / Level 2')
        ],
        ondelete="restrict"
    )
```